src/main/java/frc/robot/commands/CoralArmCommand.java

```diff
@@ -1,37 +1,52 @@
 package frc.robot.commands;
 
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
 import edu.wpi.first.wpilibj2.command.Command;
 import frc.robot.subsystems.CoralArm;
 
 public class CoralArmCommand extends Command {
 
     private final CoralArm arm;
+    private final TrapezoidProfile.Constraints constraints;
+
+    private TrapezoidProfile motionProfile;
+    private TrapezoidProfile.State motionProfileGoal;
+    private TrapezoidProfile.State motionProfileSetPoint;
 
     private double targetPositionDegrees;
     private boolean hasNewTarget;
     private boolean didReachPosition;
+    private boolean isHolding;
 
     public CoralArmCommand(CoralArm arm) {
         this.arm = arm;
+        constraints = new TrapezoidProfile.Constraints(1000,200);
 
         addRequirements(arm);
     }
 
     @Override
     public void initialize() {
-
+        isHolding = false;
     }
-
     @Override
     public void execute() {
         if (hasNewTarget) {
             // reset to move to new angle
-
-            didReachPosition = false;
-            hasNewTarget = false;
+            if(!isHolding){
+                arm.stop();
+            }else {
+
+                didReachPosition = false;
+                hasNewTarget = false;
```

Comment:
these two lines should always be updated regardless of holding or not.

`hasNewTarget` because otherwise this if will continue occuring and `didReachPosition` because we are reseting the position indication